src/v/serde/test/serde_test.cc

```diff
@@ -1077,3 +1079,39 @@ SEASTAR_THREAD_TEST_CASE(collections_interop) {
     BOOST_REQUIRE(serialized_vector == serialized_fifo);
     BOOST_REQUIRE(serialized_vector == serialized_f_vector);
 }
+
+template<typename T>
+concept IsSerdeReadable = requires(
+  iobuf_parser& in, T& t, const std::size_t bytes_left_limit) {
+    { serde::tag_invoke(serde::read_tag, in, t, bytes_left_limit) };
+};
+
+template<typename T>
+concept IsSerdeWritable = requires(T&& t, iobuf b) {
+    { serde::tag_invoke(serde::write_tag, b, std::forward<T>(t)) };
+};
+
+SEASTAR_THREAD_TEST_CASE(variant) {
+    using my_variant = serde::variant<ss::sstring, int, bool, my_enum, int>;
+    my_variant a = "foo";
+    my_variant b = my_variant(std::in_place_index_t<1>(), 1);
+    my_variant c = false;
+    my_variant d = my_enum::y;
+    my_variant e = my_variant(std::in_place_index_t<4>(), 3);
+    for (const my_variant& v : {a, b, c, d, e}) {
+        auto roundtripped = serde::from_iobuf<my_variant>(serde::to_iobuf(v));
+        BOOST_REQUIRE(v == roundtripped);
+    }
+    static_assert(
+      IsSerdeReadable<serde::variant<int>>,
+      "serde::variant **should** be compatible with serde reads");
+    static_assert(
+      IsSerdeWritable<serde::variant<int>>,
+      "serde::variant **should** be compatible with serde writes");
+    static_assert(
+      !IsSerdeReadable<std::variant<int>>,
+      "std::variant should **not** be compatible with serde reads");
+    static_assert(
+      !IsSerdeWritable<std::variant<int>>,
+      "std::variant should **not** be compatible with serde writes");
+}
```

Comment:
Mostly for exposition, consider adding a test that shows how new types can be appended to evolve the variant  

serde::variant<ss::sstring, int> -> iobuf -> serde::variant<ss::sstring, int, bool>

Reply:
Done